AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/KMSKeystoreOperations.dfy

```diff
@@ -896,4 +985,117 @@ module {:options "/functionSyntax:4" } KMSKeystoreOperations {
     && |decryptHistory.output.value.Plaintext.value| == (Structure.BKC_DIGEST_LENGTH + Structure.AES_256_LENGTH) as int
   }
 
+  method packAndCallKMS(
+    nameonly branchKeyContext: map<string, string>,
+    nameonly cryptoAndKms: CryptoAndKms,
+    nameonly material: seq<uint8>,
+    nameonly encryptionContext: map<string, string>
+  )
+    returns (output: Result<Types.EncryptedHierarchicalKey, Types.Error>)
+    requires
```

Comment:
Blocking: We need a requires HV-2.

Reply:
you technically don't.
This method is only ever called by hv-2 operations.
requiring hv-2 does not add much, whoever calls this method should prove that only on hv-2 path this gets used.

Reply:
That is exactly what a `requires` statement does!
It forces all calling code paths to be correct!